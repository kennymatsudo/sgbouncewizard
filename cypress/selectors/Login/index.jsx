const selectors = {
  loginEmailInput: "[data-test='email-field']",
  loginPasswordInput: "[data-test='password-field']",
  loginSubmitButton: "[data-test='login-button']",
  loginInvalidCredentials: "[data-test='invalid-credentials-alert']",
  loginInvalidInput: "[data-test='invalid-input-alert']",
  loginNetworkError: "[data-test='network-error-alert']",
};

export default selectors;
