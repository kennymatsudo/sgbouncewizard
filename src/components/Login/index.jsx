import React from "react";
// import { Row } from "@sendgrid/ui-components/grid/row";
// import { Column } from "@sendgrid/ui-components/grid/column";
import { TextInput } from "@sendgrid/ui-components/text-input";
import PropTypes from "prop-types";
import { Row } from "../Row";
import { Column } from "../Column";
import "./index.scss";
import InvalidCredentialsAlert from "../InvalidCredentialsAlert";
import NetworkErrorAlert from "../NetworkErrorAlert";
import InvalidInputAlert from "../InvalidInputAlert";
import LoadingButton from "../LoadingButton";
import LoginButton from "../LoginButton";

const Login = ({
  username,
  password,
  isAuthenticating,
  isAuthenticationError,
  isInvalidCredentials,
  isInvalidInput,
  isNetworkError,
  handleLoginSubmit,
  handleInputChange,
  handleAlertClose,
}) => (
  <div className="login-container">
    <Row>
      <Column width={10} offset={2}>
        <div className="login-form-container">
          <div className="login-logo">
            <img
              src="https://uiux.s3.amazonaws.com/toggleable-logos/header-logo.svg"
              alt="sg-logo"
            />
          </div>
          <Row width={8} offset={2}>
            {!isAuthenticating &&
              isAuthenticationError &&
              isInvalidCredentials && (
                <InvalidCredentialsAlert
                  data-test="invalid-credentials-alert"
                  handleAlertClose={handleAlertClose}
                />
              )}
            {!isAuthenticating &&
              isAuthenticationError &&
              isNetworkError && (
                <NetworkErrorAlert
                  data-test="network-error-alert"
                  handleAlertClose={handleAlertClose}
                />
              )}
            {!isAuthenticating &&
              isAuthenticationError &&
              isInvalidInput && (
                <InvalidInputAlert
                  data-test="invalid-input-alert"
                  handleAlertClose={handleAlertClose}
                />
              )}
          </Row>
          <div className="login-form-body">
            <Row>
              <Column width={6} offset={4}>
                <div className="input-text-wrap">
                  <TextInput
                    data-test="username-field"
                    onChange={e => handleInputChange(e, "username")}
                    value={username}
                    type="text"
                    label="Username"
                  />
                </div>
              </Column>
            </Row>
            <Row>
              <Column width={6} offset={4}>
                <TextInput
                  onChange={e => handleInputChange(e, "password")}
                  data-test="password-field"
                  value={password}
                  type="password"
                  label="Password"
                />
              </Column>
            </Row>
            <Row>
              <Column width={6} offset={4}>
                <div className="login-button-container">
                  {!isAuthenticating && (
                    <LoginButton
                      data-test="login-button"
                      handleLoginSubmit={handleLoginSubmit}
                    />
                  )}
                  {isAuthenticating && (
                    <LoadingButton
                      data-test="login-loading-button"
                      handleLoginSubmit={handleLoginSubmit}
                    />
                  )}
                </div>
              </Column>
            </Row>
          </div>
        </div>
      </Column>
    </Row>
  </div>
);

Login.propTypes = {
  username: PropTypes.string,
  password: PropTypes.string,
  isAuthenticating: PropTypes.bool,
  isAuthenticationError: PropTypes.bool,
  isInvalidCredentials: PropTypes.bool,
  isInvalidInput: PropTypes.bool,
  isNetworkError: PropTypes.bool,
  handleLoginSubmit: PropTypes.func,
  handleInputChange: PropTypes.func,
};

Login.defaultProps = {
  username: "",
  password: "",
  isAuthenticating: false,
  isAuthenticationError: false,
  isInvalidCredentials: false,
  isInvalidInput: false,
  isNetworkError: false,
  handleLoginSubmit: () => {},
  handleInputChange: () => {},
};

export default Login;
