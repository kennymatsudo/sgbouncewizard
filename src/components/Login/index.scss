.login-form-container {
  background: white;
  margin: 40px auto;
  display: inline-block;
  width: 525px;
  height: 665px;
  box-shadow: 0.5px 0 5px #9e9e9e;
}

.login-logo {
  margin-top: 9rem;
  margin-bottom: 1.5rem;
}

.login-form-body {
  text-align: left;
}

.login-button-container {
  padding-top: 25px;
  text-align: center;
}
