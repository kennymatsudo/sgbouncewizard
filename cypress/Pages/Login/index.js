import Page from "../page";
import { Selectors } from "../../../src/components/Login/selectors";

class LoginPage extends Page {
  get loginContainer() {
    return cy.get(Selectors.loginContainer);
  }

  get emailInput() {
    return cy.get(Selectors.emailInput);
  }

  get passwordInput() {
    return cy.get(Selectors.passwordInput);
  }

  get loginButton() {
    return cy.get(Selectors.loginButton);
  }

  get invalidCredentialsAlert() {
    return cy.get(Selectors.invalidCredentialsAlert);
  }

  get invalidInputAlert() {
    return cy.get(Selectors.invalidInputAlert);
  }

  get networkErrorAlert() {
    return cy.get(Selectors.networkErrorAlert);
  }

  open() {
    super.open("/");
  }

  login(email, password) {
    if (email) {
      this.emailInput.type(email);
    }

    if (password) {
      this.passwordInput.type(password);
    }

    return this.loginButton.click();
  }
}

export default new LoginPage();
