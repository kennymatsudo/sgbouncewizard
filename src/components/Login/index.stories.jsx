import React from "react";
import { storiesOf } from "@storybook/react";
import Login from "./index";

storiesOf("Login", module)
  .add("Empty Default", () => <Login />)
  .add("Some Inputs", () => (
    <Login username="testusername" password="testpassword" />
  ));
// TODO: add more storybook stories based on all the variations of props now
// We can now see all the variations of loading and error states and normal states
