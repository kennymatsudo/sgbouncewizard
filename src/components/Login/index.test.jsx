import React from "react";
import { Redirect } from "react-router-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Login, { LoginForm } from ".";

describe("Login", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Login />);
  });

  it("should render correctly", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render email input", () => {
    const loginFormWrapper = shallow(<LoginForm />);
    expect(loginFormWrapper.find('[data-test="email-field"]')).toHaveLength(1);
  });

  it("should render password input", () => {
    const loginFormWrapper = shallow(<LoginForm />);
    expect(loginFormWrapper.find('[data-test="password-field"]')).toHaveLength(
      1
    );
  });

  it("should display loading button when user is being authenticated", () => {
    const loginFormWrapper = shallow(<LoginForm />);

    loginFormWrapper.setProps({
      username: "ziv",
      password: "papa",
      isAuthenticating: true,
    });
    expect(
      loginFormWrapper.exists('[data-test="login-loading-button"]')
    ).toBeTruthy();
  });

  it("should display alert when credentials are invalid", () => {
    wrapper.setProps({
      username: "wrong",
      password: "password",
      isAuthenticating: false,
      isAuthenticationError: true,
      isInvalidCredentials: true,
    });
    expect(
      wrapper.exists('[data-test="invalid-credentials-alert"]')
    ).toBeTruthy();
    expect(wrapper.find(Redirect)).toHaveLength(0);
  });

  it("should display alert when there is a network error", () => {
    wrapper.setProps({
      username: "ziv",
      password: "papa",
      isAuthenticating: false,
      isAuthenticationError: true,
      isNetworkError: true,
    });
    expect(wrapper.exists('[data-test="network-error-alert"]')).toBeTruthy();
    expect(wrapper.find(Redirect)).toHaveLength(0);
  });

  it("should display alert when there is invalid input", () => {
    wrapper.setProps({
      username: "",
      password: "",
      isAuthenticating: false,
      isAuthenticationError: true,
      isInvalidInput: true,
    });
    expect(wrapper.exists('[data-test="invalid-input-alert"]')).toBeTruthy();
    expect(wrapper.find(Redirect)).toHaveLength(0);
  });
});
