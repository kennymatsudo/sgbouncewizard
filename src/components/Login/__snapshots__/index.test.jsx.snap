// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<div
  className="login-container"
>
  <div
    className="mako-grid__row"
  >
    <div
      className="mako-grid__col-10 col-start-2"
    >
      <div
        className="login-form-container"
      >
        <div
          className="login-logo"
        >
          <img
            alt="sg-logo"
            src="https://uiux.s3.amazonaws.com/toggleable-logos/header-logo.svg"
          />
        </div>
        <div
          className="mako-grid__row"
          offset={2}
          width={8}
        />
        <div
          className="login-form-body"
        >
          <div
            className="mako-grid__row"
          >
            <div
              className="mako-grid__col-6 col-start-4"
            >
              <div
                className="input-text-wrap"
              >
                <div
                  className="input-text-wrap"
                  style={Object {}}
                >
                  <label
                    className="input-text-label"
                  >
                    Username
                  </label>
                  <input
                    aria-describedby=""
                    data-test="username-field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    step={null}
                    type="text"
                    value=""
                  />
                  
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className="mako-grid__row"
          >
            <div
              className="mako-grid__col-6 col-start-4"
            >
              <div
                className="input-text-wrap"
                style={Object {}}
              >
                <label
                  className="input-text-label"
                >
                  Password
                </label>
                <input
                  aria-describedby=""
                  data-test="password-field"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  step={null}
                  type="password"
                  value=""
                />
                
                
              </div>
            </div>
          </div>
          <div
            className="mako-grid__row"
          >
            <div
              className="mako-grid__col-6 col-start-4"
            >
              <div
                className="login-button-container"
              >
                <button
                  className="btn btn-primary"
                  data-test="login-button"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Login
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
